protos/flyteidl/service/dataproxy.proto

```diff
@@ -136,4 +161,11 @@ service DataProxyService {
     //   description: "Creates a read-only http location that is accessible for tasks at runtime."
     // };
   }
+
+  rpc GetData (GetDataRequest) returns (GetDataResponse) {
+    // Takes an address like flyte://v1/proj/development/execid/n2/0/i and return the actual data
+    option (google.api.http) = {
+      get: "/api/v1/data/get_artifact"
```

Comment:
why do we need `get_artifact`
this is already a `GET` api
I would keep it as
`GET /api/v1/data?{data-uri}`